src/frontend-config/frontend-config.service.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { Inject, Injectable } from '@nestjs/common';
+import { ConsoleLoggerService } from '../logger/console-logger.service';
+import {
+  AuthProviders,
+  BannerDto,
+  BrandingDto,
+  CustomAuthNamesDto,
+  FrontendConfigDto,
+  IframeCommunicationDto,
+  SpecialLinksDto,
+} from './frontend-config.dto';
+import authConfiguration, { AuthConfig } from '../config/auth.config';
+import customizationConfiguration, {
+  CustomizationConfig,
+} from '../config/customization.config';
+import appConfiguration, { AppConfig } from '../config/app.config';
+import externalConfiguration, {
+  ExternalConfig,
+} from '../config/external.config';
+import { getServerVersionFromPackageJson } from '../utils/serverVersion';
+import { promises as fs, Stats } from 'fs';
+import { join } from 'path';
+
+@Injectable()
+export class FrontendConfigService {
+  constructor(
+    private readonly logger: ConsoleLoggerService,
+    @Inject(appConfiguration.KEY)
+    private appConfig: AppConfig,
+    @Inject(authConfiguration.KEY)
+    private authConfig: AuthConfig,
+    @Inject(customizationConfiguration.KEY)
+    private customizationConfig: CustomizationConfig,
+    @Inject(externalConfiguration.KEY)
+    private externalConfig: ExternalConfig,
+  ) {
+    this.logger.setContext(FrontendConfigService.name);
+  }
+
+  async getFrontendConfig(): Promise<FrontendConfigDto> {
+    return {
+      allowAnonymous: false,
```

Comment:
Still missing a few TODOs to get the actual config settings?

Reply:
added